src/org/openstreetmap/josm/gui/dialogs/relation/MemberTableModel.java

```diff
@@ -551,8 +551,8 @@ public Collection<OsmPrimitive> getSelectedChildPrimitives() {
     public Set<OsmPrimitive> getChildPrimitives(Collection<? extends OsmPrimitive> referenceSet) {
         if (referenceSet == null) return null;
         return members.stream()
-                .filter(m -> referenceSet.contains(m.getMember()))
                 .map(RelationMember::getMember)
+                .filter(referenceSet::contains)
```

Comment:
`referenceSet` is a collection. We might want to standardize and make it a set as well.

E.g.
`Collection<? extends OsmPrimitive> referenceActualSet = referenceSet instanceof Set ? (Set<? extends OsmPrimitive>) referenceSet : new HashSet<>(referenceSet);`

Reply:
Applied in 78507da4.